ui/app/components/calendar-widget.js

```diff
@@ -0,0 +1,176 @@
+import Component from '@glimmer/component';
+import layout from '../templates/components/calendar-widget';
+import { setComponentTemplate } from '@ember/component';
+import { format, formatRFC3339 } from 'date-fns';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+// ARG TODO documentation takes start and end for handQuery
+/**
+ * @module CalendarWidget
+ * CalendarWidget components are used in the client counts metrics. It helps users understand the ranges they can select.
+ *
+ * @example
+ * ```js
+ * <CalendarWidget
+ * @param {function} handleQuery - calls the parent pricing-metrics-dates handleQueryFromCalendar method which sends the data for the network request.
+ * @param {object} startDate - The start date is calculated from the parent. This component is only responsible for modifying the end Date. ANd effecting single month.
+ * />
+ *
+ * ```
+ */
+
+const MONTH_MAPPING = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+class CalendarWidget extends Component {
+  currentDate = new Date();
+  currentYear = this.currentDate.getFullYear(); // integer
+  currentMonth = parseInt(format(this.currentDate, 'M')) - 1; // integer and zero index
+
+  @tracked allMonthsNodeList = [];
+  @tracked displayYear = this.currentYear; // init to currentYear and then changes as a user clicks on the chevrons
+  @tracked disablePastYear = this.isObsoleteYear(); // if obsolete year, disable left chevron
+  @tracked disableFutureYear = this.isCurrentYear(); // if current year, disable right chevron
+  @tracked endDateDisplay;
+  @tracked showCalendar = false;
+  @tracked showSingleMonth = false;
+
+  constructor() {
+    super(...arguments);
+    // ARG TODO we need to return the config's duration if not default to 12 months to calculate, now it's 12 months
+    let date = new Date();
```

Comment:
right now I'm calculating the startDate based on today, but the next PR will pull the startDate from the license endpoint and send it to the component as a param. This component is only in charge of endDates or singleMonth or saying, just go back to the default billing period.